Index: scripts/eastern_kingdoms/scarlet_enclave/ebon_hold.cpp
===================================================================
--- scripts/eastern_kingdoms/scarlet_enclave/ebon_hold.cpp	(revision 1749)
+++ scripts/eastern_kingdoms/scarlet_enclave/ebon_hold.cpp	(working copy)
@@ -1100,6 +1100,90 @@
     return false;
 }
 
+/*######
+## npc_eye_of_acherus
+######*/
+
+struct MANGOS_DLL_DECL npc_eye_of_acherusAI : public ScriptedAI
+{
+    npc_eye_of_acherusAI(Creature *pCreature) : ScriptedAI(pCreature)
+    {
+        m_creature->SetActiveObjectState(true);
+        m_creature->SetLevel(55); //else one hack
+        StartTimer = 2000;
+        Active = false;
+    }
+
+    uint32 StartTimer;
+    bool Active;
+
+    void Reset(){}
+    void AttackStart(Unit *) {}
+    void MoveInLineOfSight(Unit*) {}
+
+    void JustDied(Unit*u)
+    {
+        if(!m_creature || m_creature->GetTypeId() != TYPEID_UNIT)
+            return;
+
+        Unit *target = m_creature->GetCharmer();
+
+        if(!target || target->GetTypeId() != TYPEID_PLAYER)
+            return;
+
+        m_creature->SetCharmerGUID(0);
+        target->RemoveAurasDueToSpell(51852);
+        target->SetCharm(NULL);
+
+        ((Player*)target)->GetCamera().ResetView();
+        ((Player*)target)->SetClientControl(m_creature,0);
+        ((Player*)target)->SetMover(NULL);
+
+        m_creature->CleanupsBeforeDelete();
+        m_creature->AddObjectToRemoveList();
+        //m_creature->ForcedDespawn();
+            return;
+    }
+
+    void MovementInform(uint32 uiType, uint32 uiPointId)
+    {
+        if (uiType != POINT_MOTION_TYPE && uiPointId == 0)
+            return;
+
+            char * text = "The Eye of Acherus is in your control";
+            m_creature->MonsterTextEmote(text, m_creature->GetGUID(), true);
+            m_creature->CastSpell(m_creature, 51890, true);
+    }
+
+    void UpdateAI(const uint32 uiDiff)
+    {
+        if(m_creature->isCharmed())
+        {
+            if (StartTimer < uiDiff && !Active)
+            {
+                m_creature->CastSpell(m_creature, 70889, true);
+                m_creature->CastSpell(m_creature, 51892, true);
+                char * text = "The Eye of Acherus launches towards its destination";
+                m_creature->MonsterTextEmote(text, m_creature->GetGUID(), true);
+                m_creature->SetSpeedRate(MOVE_FLIGHT, 6.4f,true);
+                m_creature->GetMotionMaster()->MovePoint(0, 1750.8276f, -5873.788f, 147.2266f);
+                Active = true;
+            }
+            else StartTimer -= uiDiff;
+        }
+        else
+        {
+            m_creature->CleanupsBeforeDelete();
+            m_creature->AddObjectToRemoveList();
+        }
+    }
+};
+
+CreatureAI* GetAI_npc_eye_of_acherus(Creature* pCreature)
+{
+    return new npc_eye_of_acherusAI(pCreature);
+}
+
 void AddSC_ebon_hold()
 {
     Script *newscript;
@@ -1136,4 +1220,9 @@
     newscript->Name = "go_acherus_soul_prison";
     newscript->pGOHello = &GOHello_go_acherus_soul_prison;
     newscript->RegisterSelf();
+
+    newscript = new Script;
+    newscript->Name = "npc_eye_of_acherus";
+    newscript->GetAI = &GetAI_npc_eye_of_acherus;
+    newscript->RegisterSelf();
 }
